src/main/java/org/zone/config/node/title/DefaultTitleFadeInNode.java

```diff
@@ -0,0 +1,113 @@
+package org.zone.config.node.title;
+
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.zone.ZonePlugin;
+import org.zone.commands.system.CommandArgument;
+import org.zone.commands.system.arguments.simple.EnumArgument;
+import org.zone.commands.system.arguments.simple.number.IntegerArgument;
+import org.zone.commands.system.context.CommandContext;
+import org.zone.config.command.ConfigCommandNode;
+import org.zone.config.node.ZoneNode;
+import org.zone.utils.Messages;
+import org.zone.utils.time.TimeUnits;
+import org.zone.utils.time.duration.TimeDuration;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+
+public class DefaultTitleFadeInNode implements ZoneNode.WithDefault<TimeDuration> {
+
+    private class DefaultTitleFadeInConfigCommandTimeNode implements ConfigCommandNode<Integer> {
+
+        @Override
+        public String getDisplayId() {
+            return "zone.message.title.fadeIn.length";
+        }
+
+        @Override
+        public CommandArgument<Integer> getCommandArgument() {
+            return new IntegerArgument("length");
+        }
+
+        @Override
+        public CommandResult onChange(
+                CommandContext context, Integer newValue) {
+            try {
+                DefaultTitleFadeInNode.this.set(ZonePlugin.getZonesPlugin().getConfig(),
+                        new TimeDuration(DefaultTitleFadeInNode.this.getInitialValue().getTimeUnit(),
+                                newValue));
+                return CommandResult.success();
+            } catch (SerializationException se) {
+                se.printStackTrace();
+                return CommandResult.error(Messages.getZoneSavingError(se));
+            }
+        }
+    }
+
+    private class DefaultTitleFadeInConfigCommandTimeUnitsNode implements ConfigCommandNode<TimeUnits> {
+
+        @Override
+        public String getDisplayId() {
+            return "zone.message.title.fadeIn.timeUnit";
+        }
+
+        @Override
+        public CommandArgument<TimeUnits> getCommandArgument() {
+            return new EnumArgument<>("timeUnit", TimeUnits.class);
+        }
+
+        @Override
+        public CommandResult onChange(CommandContext context, TimeUnits newValue) {
+            try {
+                DefaultTitleFadeInNode.this.set(ZonePlugin.getZonesPlugin().getConfig(),
+                        new TimeDuration(newValue, 1));
```

Comment:
when a user changes the unit, the length is reset to 1? You sure thats how you want to play it?

Reply:
no